### DISCUSSION TITLE
Scrolling in the editor is extremely unresponsive in lazer on macOS

### DISCUSSION THREAD

**Discussion by hrfarmer** [ppy/osu#27530] [Ideas]
I'm surprised I haven't seen much, if any feedback on this, but I find that scrolling through the timeline on lazer is VERY unresponsive compared to stable. (see attached video)
I read in a discussion (#23818) that there was some type of acceleration for scrolling, which if that's the case here, i *really* think that it should not apply in the editor, as it makes precise scrolling *extremely* difficult. It's really the only major thing that's stopping me from using it more over stable.

EDIT: I tried scrolling on my windows laptop with the same mouse, and the same logitech mouse drivers running, and it turns out the scrolling works as expected there. This might just be a macOS specific issue.

https://github.com/ppy/osu/assets/58487401/f2d35693-8632-414d-a688-f12b0388dd3c

**Comment by cihe13375** [comment]
So from the video it seems that in lazer the timeline only moves after multiple scrolls, right?
Does the mouse has some kind of "infinite scrolling" (i.e. the wheel moves continuously instead of in "steps")?

**Reply by hrfarmer** [reply]
It does, but I have the scroll wheel set to move in steps instead of infinite scroll. I also have smooth scrolling off in logitech settings, so I'm fairly certain the hardware isn't the issue here.

**Reply by cihe13375** [reply]
interesting. I don't have macOS though so cannot say much. Maybe try turn on/off high precision mouse in settings; if it doesn't work I have no idea.

**Comment by peppy** [comment]
This does look like a driver/configuration issue. I'm guessing logitech is trying to do their own "smooth scrolling" implementation but it's not working well with your expectations of a non-smooth scroll.

You could adjust your drivers and turn off smooth scrolling, or just scroll the wheel faster.

**Reply by hrfarmer** [reply]
I did some additional testing and I think this could just be an issue with how macOS is handling the scrolling from this mouse (MX Master 3S). I have always had smooth scrolling off in the drivers, and I even completely killed the drivers in activity monitor, but I still had the scrolling issue. However, I plugged in a g pro wireless to my macbook (which there's no drivers for on my macbook), and the scrolling is completely normal as expected. Additionally, I tried using my mx master on my windows laptop, *with* the same logitech drivers running, and the scrolling was normal on it.

I'm fairly certain this is some type of acceleration issue though, because when I turn off scroll acceleration on the mx master using [LinearMouse](https://linearmouse.app/), the scrolling will then work as expected.

TLDR This is probably just an issue with the mouse and/or macOS, so I doubt anything could be done on the lazer side to fix this.
